futures-util/src/stream/futures_unordered/iter_mut.rs

```diff
@@ -0,0 +1,34 @@
+use std::marker::PhantomData;
+use super::FuturesUnordered;
+use super::node::Node;
+
+#[derive(Debug)]
+/// Mutable iterator over all futures in the unordered set.
```

Comment:
Subtle: this is unsound now because it gives mutable access to the futures allowing them to be `mem::replaced`, although they may have been polled. This should either yield `PinMut<F>` or `iter_mut` should be changed to include an `F: Unpin` bound. Perhaps offer both as `iter_mut` and `iter_pin_mut`?

Reply:
Good catch! Done.